libs/data-access/api/src/lib/trpc/routers/enrollment.router.ts

```diff
@@ -34,6 +35,54 @@ export const enrollmentRouter = t.router({
 		})
 });
 
+export async function getEnrollmentsOfUserWithCourseDetails(
+	username: string
+): Promise<EnrollmentWithDetails[]> {
+	const enrollments = await database.enrollment.findMany({
+		where: { username },
+		select: {
+			completedAt: true,
+			status: true,
+			lastProgressUpdate: true,
+			course: {
+				select: {
+					title: true,
+					slug: true,
+					imgUrl: true,
+					authors: {
+						select: {
+							displayName: true
+						}
+					}
+				}
+			}
+		}
+	});
+
+	const enrollmentsWithDetails = await Promise.all(
+		enrollments.map(async enrollment => {
+			const completions = await getCourseCompletionOfStudent(
+				enrollment.course.slug,
+				username
+			);
+
+			return {
+				...enrollment,
+				completedAt: enrollment.completedAt ? enrollment.completedAt.toISOString() : null,
+				lastProgressUpdate: enrollment.lastProgressUpdate.toISOString(),
+				completions
+			};
+		})
+	);
+
+	enrollmentsWithDetails.sort(
+		(a, b) =>
+			new Date(b.lastProgressUpdate).getTime() - new Date(a.lastProgressUpdate).getTime()
+	);
+
+	return enrollmentsWithDetails as EnrollmentWithDetails[];
+}
```

Comment:
wenn das nicht zur wiederverwendung gedacht ist, sollte das in die datei wo auch die ServerSideProps geladen werden. 

im Router sollte das schon wenn möglich router bezogen sein. Es war mal eine notdürftige hilfe select methods zur wiederverwendung (da einmal auf server side und einmal auf client side benötigt) in die router lib zu schmeißen.

Reply:
Wird nicht wieder verwendet, habe das in die Site verschoben.